test.py

```diff
@@ -1432,11 +1432,19 @@ async def find_tests(options: argparse.Namespace) -> None:
     # else only specified suites
     suites = ["*"] if not suite_names or None in suite_names else suite_names
     suite_confs = [f for s in suites for f in glob.glob(os.path.join("test", s, "suite.yaml"))]
+    add_tests_tasks = set()
     for cfg_file in suite_confs:
         suite_dir = os.path.dirname(cfg_file)            # Suite directory
         for mode in options.modes:
             suite = TestSuite.opt_create(suite_dir, options, mode)
-            await suite.add_test_list()
+            add_tests_tasks.add(suite.add_test_list())
+    try:
+        await asyncio.gather(*add_tests_tasks)
+    except asyncio.CancelledError:
+        for task in add_tests_tasks:
+            task.cancel()
+        await asyncio.gather(*add_tests_tasks, return_exceptions=True)
```

Comment:
"Improves 40 to 20 seconds"?!

```
time test/cql-pytest/run test_json.py::test_fromjson_timestamp_submilli
```
Currently takes 1 second (0.3 seconds to boot Scylla, 0.17 seconds to run the test, and about half a second more for I am not sure exactly what).

Are you saying that if I use test.py to run a single test it will take 20 seconds (at least not 40 seconds!) just to figure out what I meant?

I think that you must have a special case for running a single test case or a single test file. This special case should NOT try to discover all other tests.